sklearn/model_selection/_validation.py

```diff
@@ -231,12 +235,16 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
             estimator.fit(X_train, **fit_params)
         else:
             estimator.fit(X_train, y_train, **fit_params)
+        fit_time = time.time() - start_time
 
     except Exception as e:
+        # Note fit time as time until error
+        fit_time = time.time() - start_time
         if error_score == 'raise':
             raise
         elif isinstance(error_score, numbers.Number):
             test_score = error_score
+            score_time = time.time() - fit_time
```

Comment:
Yes... it will not be negative...

The way the exception handling is done, overall time cannot be calculated outside the try...except...else... loop as training scores will be included in timing. But I've now set `score_time` to 0 and added a test for timing.